<div align="right" style=" font-size: 80%; text-align: center; margin: 0 auto">
<img
 src="https://raw.githubusercontent.com/Explore-AI/Pictures/master/alx-courses/aice/assets/Content_page_banner_blue_dots.png"
 alt="ALX Content Header"
 class="full-width-image"
/>
</div>

# Logical and comparison operators ii

> ⚠️ This notebook will not run on Google Colab because it cannot connect to a local database. Please make sure that this notebook is running on the same local machine as your MySQL Workbench installation and MySQL `united_nations` database.

## Learning objectives

In this train we will learn:
- How to use the IS NULL and IS NOT NULL statements
- How to use the IS IN and IS NOT IN statements
- Understand if there is any correlation between the GDP and availibility of drinking water and sanitation services in Sub-Saharan Africa

## Connecting to our MySQL database

Using our `Access_to_Basic_Services` table in our `united_nations` database we created in MySQL Workbench, we want to answer some questions about our dataset. We can apply the same queries we used in MySQL Workbench in this notebook if we connect to our MySQL server by running the cells below.


In [1]:
# Load and activate the SQL extension to allows us to execute SQL in a Jupyter notebook. 
# If you get an error here, make sure that mysql and pymysql is installed correctly. 

%load_ext sql

In [54]:
# Establish a connection to the local database using the '%sql' magic command,
# Replace 'password' with our connection password and `db_name` with our database name. 
# If you get an error here, please make sure the database name or password is correct.

%sql mysql+pymysql://root:12122001UPi@localhost:3306/united_nations

Connecting and switching to connection 'mysql+pymysql://root:***@localhost:3306/united_nations'

In [55]:
%config SqlMagic.displaylimit = None;

displaylimit: Value None will be treated as 0 (no limit)


To make a query, we add the `%%sql` command to the start of a cell, create one open line and then the query like below and run the cell.

In [56]:
%%sql
SHOW TABLES;

Running query in 'mysql+pymysql://root:***@localhost:3306/united_nations'

1 rows affected.

+--------------------------+
| Tables_in_united_nations |
+--------------------------+
| access_to_basic_services |
+--------------------------+

In [57]:
%%sql

SELECT 
    *
FROM
    united_nations.Access_to_Basic_Services
LIMIT 5;

Running query in 'mysql+pymysql://root:***@localhost:3306/united_nations'

5 rows affected.

+---------------------------+--------------+--------------+-------------+-------------------------------------+---------------------------------+----------------------------+---------------------+------------+------------------+
|           Region          |  Sub_region  | Country_name | Time_period | Pct_managed_drinking_water_services | Pct_managed_sanitation_services | Est_population_in_millions | Est_gdp_in_billions | Land_area  | Pct_unemployment |
+---------------------------+--------------+--------------+-------------+-------------------------------------+---------------------------------+----------------------------+---------------------+------------+------------------+
| Central and Southern Asia | Central Asia |  Kazakhstan  |     2015    |                94.67                |              98.00              |         17.542806          |        184.39       | 2699700.00 |       4.93       |
| Central and Southern Asia | Central Asia |  Kazakhstan  |     2016    |                94.67                |              98.00              |         17.794055          |        137.28       | 2699700.00 |       4.96       |
| Central and Southern Asia | Central Asia |  Kazakhstan  |     2017    |                95.00                |              98.00              |         18.037776          |        166.81       | 2699700.00 |       4.90       |
| Central and Southern Asia | Central Asia |  Kazakhstan  |     2018    |                95.00                |              98.00              |         18.276452          |        179.34       | 2699700.00 |       4.85       |
| Central and Southern Asia | Central Asia |  Kazakhstan  |     2019    |                95.00                |              98.00              |         18.513673          |        181.67       | 2699700.00 |       4.80       |
+---------------------------+--------------+--------------+-------------+-------------------------------------+---------------------------------+----------------------------+---------------------+------------+------------------+

In [58]:
%%sql

SELECT 
    *
FROM
    Access_to_Basic_Services
Limit 10;

Running query in 'mysql+pymysql://root:***@localhost:3306/united_nations'

10 rows affected.

+---------------------------+--------------+--------------+-------------+-------------------------------------+---------------------------------+----------------------------+---------------------+------------+------------------+
|           Region          |  Sub_region  | Country_name | Time_period | Pct_managed_drinking_water_services | Pct_managed_sanitation_services | Est_population_in_millions | Est_gdp_in_billions | Land_area  | Pct_unemployment |
+---------------------------+--------------+--------------+-------------+-------------------------------------+---------------------------------+----------------------------+---------------------+------------+------------------+
| Central and Southern Asia | Central Asia |  Kazakhstan  |     2015    |                94.67                |              98.00              |         17.542806          |        184.39       | 2699700.00 |       4.93       |
| Central and Southern Asia | Central Asia |  Kazakhstan  |     2016    |                94.67                |              98.00              |         17.794055          |        137.28       | 2699700.00 |       4.96       |
| Central and Southern Asia | Central Asia |  Kazakhstan  |     2017    |                95.00                |              98.00              |         18.037776          |        166.81       | 2699700.00 |       4.90       |
| Central and Southern Asia | Central Asia |  Kazakhstan  |     2018    |                95.00                |              98.00              |         18.276452          |        179.34       | 2699700.00 |       4.85       |
| Central and Southern Asia | Central Asia |  Kazakhstan  |     2019    |                95.00                |              98.00              |         18.513673          |        181.67       | 2699700.00 |       4.80       |
| Central and Southern Asia | Central Asia |  Kazakhstan  |     2020    |                95.00                |              98.00              |         18.755666          |        171.08       | 2699700.00 |       4.89       |
| Central and Southern Asia | Central Asia |  Kyrgyzstan  |     2015    |                89.67                |              96.67              |            None            |         None        |    None    |       None       |
| Central and Southern Asia | Central Asia |  Kyrgyzstan  |     2016    |                90.33                |              96.67              |            None            |         None        |    None    |       None       |
| Central and Southern Asia | Central Asia |  Kyrgyzstan  |     2017    |                91.00                |              97.33              |            None            |         None        |    None    |       None       |
| Central and Southern Asia | Central Asia |  Kyrgyzstan  |     2018    |                91.33                |              97.33              |            None            |         None        |    None    |       None       |
+---------------------------+--------------+--------------+-------------+-------------------------------------+---------------------------------+----------------------------+---------------------+------------+------------------+

## Exercise


We will be working with the `united_nations.Access_to_Basic_Services` table, which contains information about different countries, their access to basic services, and their estimated GDP.

In this exercise We would like to determine if the GDP of a country, specifically in Sub-Saharan Africa has any correlation to its access to basic services


### Task 1 
Select data from the Sub-Saharan African region during the year 2020.

In [53]:
%%sql
SELECT * FROM Access_to_Basic_Services WHERE Region = 'Sub-Saharan Africa' AND Time_period = 2020 LIMIT 10;

Running query in 'mysql+pymysql://root:***@localhost:3306/united_nations'

10 rows affected.

+--------------------+----------------+--------------+-------------+---------------------+---------------------------------+----------------------------+---------------------+------------+------------------+
|       Region       |   Sub_region   | Country_name | Time_period | pct_access_to_water | Pct_managed_sanitation_services | Est_population_in_millions | Est_gdp_in_billions | Land_area  | Pct_unemployment |
+--------------------+----------------+--------------+-------------+---------------------+---------------------------------+----------------------------+---------------------+------------+------------------+
| Sub-Saharan Africa | Eastern Africa |   Burundi    |     2020    |        70.33        |              44.33              |         12.220227          |         2.65        |  25680.00  |       1.03       |
| Sub-Saharan Africa | Eastern Africa |   Djibouti   |     2020    |        69.00        |              56.00              |          1.090156          |         3.18        |  23180.00  |       None       |
| Sub-Saharan Africa | Eastern Africa |   Ethiopia   |     2020    |        58.00        |              11.67              |         117.190911         |        107.66       | 1128571.26 |       None       |
| Sub-Saharan Africa | Eastern Africa |    Kenya     |     2020    |        67.00        |              33.67              |         51.985780          |        100.67       | 569140.00  |       None       |
| Sub-Saharan Africa | Eastern Africa |  Madagascar  |     2020    |        56.33        |              13.00              |         28.225177          |        13.05        | 581800.00  |       None       |
| Sub-Saharan Africa | Eastern Africa |    Malawi    |     2020    |        74.33        |              28.67              |         19.377061          |        12.18        |  94280.00  |       0.91       |
| Sub-Saharan Africa | Eastern Africa |  Mauritius   |     2020    |        100.00       |              96.00              |          1.265740          |        11.40        |  2030.00   |       8.63       |
| Sub-Saharan Africa | Eastern Africa |   Mayotte    |     2020    |        96.00        |              100.00             |            None            |         None        |    None    |       None       |
| Sub-Saharan Africa | Eastern Africa |  Mozambique  |     2020    |        66.67        |              40.33              |         31.178239          |        14.03        | 786380.00  |       None       |
| Sub-Saharan Africa | Eastern Africa |    Rwanda    |     2020    |        66.33        |              64.00              |         13.146362          |        10.18        |  24670.00  |      11.83       |
+--------------------+----------------+--------------+-------------+---------------------+---------------------------------+----------------------------+---------------------+------------+------------------+

### Task 2

Sometimes there are null values in our entries. Any country having Null values for their GDP should not be included in our query as they will not help us determine if there is any correlation between GDP and access to basic services. For this task determine if there are any NULL values in the GDP column

In [29]:
%%sql
    SELECT * FROM Access_to_Basic_Services  WHERE Region = 'Sub-Saharan Africa' AND Est_gdp_in_billions IS NULL LIMIT 5;
# Add your code here

Running query in 'mysql+pymysql://root:***@localhost:3306/united_nations'

5 rows affected.

+---------------------------+--------------+--------------+-------------+---------------------+---------------------------------+----------------------------+---------------------+-----------+------------------+
|           Region          |  Sub_region  | Country_name | Time_period | pct_access_to_water | Pct_managed_sanitation_services | Est_population_in_millions | Est_gdp_in_billions | Land_area | Pct_unemployment |
+---------------------------+--------------+--------------+-------------+---------------------+---------------------------------+----------------------------+---------------------+-----------+------------------+
| Central and Southern Asia | Central Asia |  Kyrgyzstan  |     2015    |        89.67        |              96.67              |            None            |         None        |    None   |       None       |
| Central and Southern Asia | Central Asia |  Kyrgyzstan  |     2016    |        90.33        |              96.67              |            None            |         None        |    None   |       None       |
| Central and Southern Asia | Central Asia |  Kyrgyzstan  |     2017    |        91.00        |              97.33              |            None            |         None        |    None   |       None       |
| Central and Southern Asia | Central Asia |  Kyrgyzstan  |     2018    |        91.33        |              97.33              |            None            |         None        |    None   |       None       |
| Central and Southern Asia | Central Asia |  Kyrgyzstan  |     2019    |        91.67        |              97.33              |            None            |         None        |    None   |       None       |
+---------------------------+--------------+--------------+-------------+---------------------+---------------------------------+----------------------------+---------------------+-----------+------------------+

### Task 3

If there are any Null values exclude them from your query.

In [80]:
%%sql
SELECT * FROM Access_to_Basic_Services  WHERE Region = 'Sub-Saharan Africa' AND Time_period = 2020 AND Est_gdp_in_billions IS NOT NULL ;

Running query in 'mysql+pymysql://root:***@localhost:3306/united_nations'

38 rows affected.

+--------------------+-----------------+--------------------------+-------------+-------------------------------------+---------------------------------+----------------------------+---------------------+------------+------------------+
|       Region       |    Sub_region   |       Country_name       | Time_period | Pct_managed_drinking_water_services | Pct_managed_sanitation_services | Est_population_in_millions | Est_gdp_in_billions | Land_area  | Pct_unemployment |
+--------------------+-----------------+--------------------------+-------------+-------------------------------------+---------------------------------+----------------------------+---------------------+------------+------------------+
| Sub-Saharan Africa |  Eastern Africa |         Burundi          |     2020    |                70.33                |              44.33              |         12.220227          |         2.65        |  25680.00  |       1.03       |
| Sub-Saharan Africa |  Eastern Africa |         Djibouti         |     2020    |                69.00                |              56.00              |          1.090156          |         3.18        |  23180.00  |       None       |
| Sub-Saharan Africa |  Eastern Africa |         Ethiopia         |     2020    |                58.00                |              11.67              |         117.190911         |        107.66       | 1128571.26 |       None       |
| Sub-Saharan Africa |  Eastern Africa |          Kenya           |     2020    |                67.00                |              33.67              |         51.985780          |        100.67       | 569140.00  |       None       |
| Sub-Saharan Africa |  Eastern Africa |        Madagascar        |     2020    |                56.33                |              13.00              |         28.225177          |        13.05        | 581800.00  |       None       |
| Sub-Saharan Africa |  Eastern Africa |          Malawi          |     2020    |                74.33                |              28.67              |         19.377061          |        12.18        |  94280.00  |       0.91       |
| Sub-Saharan Africa |  Eastern Africa |        Mauritius         |     2020    |                100.00               |              96.00              |          1.265740          |        11.40        |  2030.00   |       8.63       |
| Sub-Saharan Africa |  Eastern Africa |        Mozambique        |     2020    |                66.67                |              40.33              |         31.178239          |        14.03        | 786380.00  |       None       |
| Sub-Saharan Africa |  Eastern Africa |          Rwanda          |     2020    |                66.33                |              64.00              |         13.146362          |        10.18        |  24670.00  |      11.83       |
| Sub-Saharan Africa |  Eastern Africa |         Somalia          |     2020    |                57.33                |              40.00              |         16.537016          |         6.88        | 627340.00  |       None       |
| Sub-Saharan Africa |  Eastern Africa |          Uganda          |     2020    |                61.00                |              21.67              |         44.404611          |        37.60        | 200520.00  |       None       |
| Sub-Saharan Africa |  Eastern Africa |          Zambia          |     2020    |                66.67                |              32.67              |         18.927715          |        18.11        | 743390.00  |       6.03       |
| Sub-Saharan Africa |  Eastern Africa |         Zimbabwe         |     2020    |                68.00                |              36.33              |         15.669666          |        21.51        | 386850.00  |       None       |
| Sub-Saharan Africa |  Middle Africa  |          Angola          |     2020    |                52.33                |              47.00              |         33.428486          |        53.62        | 124

### Task 4

Lets get an idea if there's any correlation between the GDP and access to basic services for the top 5 economies in Sub-Saharan Africr. The top 5 GDP's are : ('Nigeria','South Africa','Ethiopia','Kenya','Ghana'). Make sure your query only includes these countries.

In [81]:
%%sql
# Add your code here
SELECT 
Country_name, Est_gdp_in_billions, Pct_managed_drinking_water_services, Est_population_in_millions	
FROM 
Access_to_Basic_Services
WHERE 
Time_period = 2020 
AND 
Region = 'Sub-Saharan Africa'
AND
Country_name IN  ('Nigeria','South Africa','Ethiopia','Kenya','Ghana')
AND 
 Est_gdp_in_billions IS NOT NULL 
ORDER BY Est_gdp_in_billions DESC;

Running query in 'mysql+pymysql://root:***@localhost:3306/united_nations'

5 rows affected.

+--------------+---------------------+-------------------------------------+----------------------------+
| Country_name | Est_gdp_in_billions | Pct_managed_drinking_water_services | Est_population_in_millions |
+--------------+---------------------+-------------------------------------+----------------------------+
|   Nigeria    |        432.20       |                77.33                |         208.327405         |
| South Africa |        337.62       |                92.00                |         58.801927          |
|   Ethiopia   |        107.66       |                58.00                |         117.190911         |
|    Kenya     |        100.67       |                67.00                |         51.985780          |
|    Ghana     |        70.04        |                84.67                |         32.180401          |
+--------------+---------------------+-------------------------------------+----------------------------+

Based on your results. Is there any correlation between the GDP and access to basic services? Does a higher GDP translate to better services?

### Task 5

We only looked at 5 countries in the previous query. Lets have a look at the rest of Sub-Saharan Africa .
Exclude the countries mentioned in the previous task

In [82]:
%%sql
SELECT 
Country_name, Est_gdp_in_billions, Pct_managed_drinking_water_services, Est_population_in_millions
FROM 
Access_to_Basic_Services
WHERE  
Time_period = 2020 
AND 
Region = 'Sub-Saharan Africa'
AND
Country_name 
NOT IN  ('Nigeria','South Africa','Ethiopia','Kenya','Ghana') ORDER BY Pct_managed_drinking_water_services DESC
;

Running query in 'mysql+pymysql://root:***@localhost:3306/united_nations'

42 rows affected.

+----------------------------------+---------------------+-------------------------------------+----------------------------+
|           Country_name           | Est_gdp_in_billions | Pct_managed_drinking_water_services | Est_population_in_millions |
+----------------------------------+---------------------+-------------------------------------+----------------------------+
|            Mauritius             |        11.40        |                100.00               |          1.265740          |
|             Réunion              |         None        |                100.00               |            None            |
|           Saint Helena           |         None        |                99.00                |            None            |
|             Mayotte              |         None        |                96.00                |            None            |
|             Botswana             |        14.93        |                89.67                |          2.546402          |
|            Cabo Verde            |         1.70        |                87.33                |          0.582640          |
|             Senegal              |        24.49        |                85.00                |         16.436120          |
|             Namibia              |        10.58        |                83.67                |          2.489098          |
|               Mali               |        17.47        |                83.67                |         21.224040          |
|              Gambia              |         None        |                79.33                |            None            |
|      Sao Tome and Principe       |         0.47        |                77.33                |          0.218641          |
|             Eswatini             |         3.98        |                76.67                |          1.180655          |
|             Lesotho              |         2.23        |                76.33                |          2.254100          |
|             Liberia              |         3.04        |                75.00                |          5.087584          |
|              Malawi              |        12.18        |                74.33                |         19.377061          |
|              Gabon               |        15.31        |                73.33                |          2.292573          |
|          Côte d'Ivoire           |         None        |                70.67                |            None            |
|               Togo               |         7.57        |                70.67                |          8.442580          |
|             Burundi              |         2.65        |                70.33                |         12.220227          |
|            Mauritania            |         8.41        |                70.33                |          4.498604          |
|             Djibouti             |         3.18        |                69.00                |          1.090156          |
|              Congo               |         None        |                69.00                |            None            |
|             Zimbabwe             |        21.51        |                68.00                |         15.669666          |
|              Guinea              |        14.18        |                67.33                |         13.205153          |
|            Mozambique            |        14.03        |                66.67                |         31.178239          |
|              Zambia              |        18.11        |                66.67                |         18.927715          |
|              Rwanda              |        10.18        |                66.33                |         13.146362          |
|              Benin               |        15.65        |                65.33                |         12.643123          |
|   United Republic of Tanzania    |         None        |                65.00               

Again. Do you see any correlation? Arrange the `Pct_managed_drinking_water_services` in descending order. Do the countries with the highest percentage to `Pct_managed_drinking_water_services` have a higher GDP then the countries with a lower percentage of `Pct_managed_drinking_water_services` ?
 Now arrange the `Pct_managed_sanitation_services` in descending order. Do countries with a higher `Pct_managed_sanitation_services` also have a higher GDP?

## Solutions

### Task 1

In [61]:
%%sql

SELECT 
	Country_name,
	Time_period,
	Pct_managed_drinking_water_services,
	Pct_managed_sanitation_services,
	Est_population_in_millions,
	Est_gdp_in_billions
FROM 
	united_nations.Access_to_Basic_Services
WHERE
	region = 'Sub-Saharan Africa'
AND 
    Time_period = 2020;

Running query in 'mysql+pymysql://root:***@localhost:3306/united_nations'

47 rows affected.

+----------------------------------+-------------+-------------------------------------+---------------------------------+----------------------------+---------------------+
|           Country_name           | Time_period | Pct_managed_drinking_water_services | Pct_managed_sanitation_services | Est_population_in_millions | Est_gdp_in_billions |
+----------------------------------+-------------+-------------------------------------+---------------------------------+----------------------------+---------------------+
|             Burundi              |     2020    |                70.33                |              44.33              |         12.220227          |         2.65        |
|             Djibouti             |     2020    |                69.00                |              56.00              |          1.090156          |         3.18        |
|             Ethiopia             |     2020    |                58.00                |              11.67              |         117.190911         |        107.66       |
|              Kenya               |     2020    |                67.00                |              33.67              |         51.985780          |        100.67       |
|            Madagascar            |     2020    |                56.33                |              13.00              |         28.225177          |        13.05        |
|              Malawi              |     2020    |                74.33                |              28.67              |         19.377061          |        12.18        |
|            Mauritius             |     2020    |                100.00               |              96.00              |          1.265740          |        11.40        |
|             Mayotte              |     2020    |                96.00                |              100.00             |            None            |         None        |
|            Mozambique            |     2020    |                66.67                |              40.33              |         31.178239          |        14.03        |
|              Rwanda              |     2020    |                66.33                |              64.00              |         13.146362          |        10.18        |
|             Réunion              |     2020    |                100.00               |              100.00             |            None            |         None        |
|             Somalia              |     2020    |                57.33                |              40.00              |         16.537016          |         6.88        |
|           South Sudan            |     2020    |                48.33                |              22.33              |         10.606227          |         None        |
|              Uganda              |     2020    |                61.00                |              21.67              |         44.404611          |        37.60        |
|   United Republic of Tanzania    |     2020    |                65.00                |              34.00              |            None            |         None        |
|              Zambia              |     2020    |                66.67                |              32.67              |         18.927715          |        18.11        |
|             Zimbabwe             |     2020    |                68.00                |              36.33              |         15.669666          |        21.51        |
|              Angola              |     2020    |                52.33                |              47.00              |         33.428486          |        53.62        |
|             Cameroon             |     2020    |                64.00                |              43.00              |         26.491087          |        40.77        |
|     Central African Republic     |     2020    |                38.33                |              15.00              |          5.343020          |         2.33        

### Task 2

In [59]:
%%sql

SELECT 
	Country_name,
	Time_period,
	Pct_managed_drinking_water_services,
	Pct_managed_sanitation_services,
	Est_gdp_in_billions,
    region
FROM 
	united_nations.Access_to_Basic_Services
WHERE
	region = 'Sub-Saharan Africa'
AND 
	Time_period = 2020
AND 
	Est_gdp_in_billions IS NULL;

# Use LIMIT if you think the results set will be large



Running query in 'mysql+pymysql://root:***@localhost:3306/united_nations'

9 rows affected.

+----------------------------------+-------------+-------------------------------------+---------------------------------+---------------------+--------------------+
|           Country_name           | Time_period | Pct_managed_drinking_water_services | Pct_managed_sanitation_services | Est_gdp_in_billions |       region       |
+----------------------------------+-------------+-------------------------------------+---------------------------------+---------------------+--------------------+
|             Mayotte              |     2020    |                96.00                |              100.00             |         None        | Sub-Saharan Africa |
|             Réunion              |     2020    |                100.00               |              100.00             |         None        | Sub-Saharan Africa |
|           South Sudan            |     2020    |                48.33                |              22.33              |         None        | Sub-Saharan Africa |
|   United Republic of Tanzania    |     2020    |                65.00                |              34.00              |         None        | Sub-Saharan Africa |
|              Congo               |     2020    |                69.00                |              17.67              |         None        | Sub-Saharan Africa |
| Democratic Republic of the Congo |     2020    |                47.67                |              15.33              |         None        | Sub-Saharan Africa |
|          Côte d'Ivoire           |     2020    |                70.67                |              34.67              |         None        | Sub-Saharan Africa |
|              Gambia              |     2020    |                79.33                |              44.33              |         None        | Sub-Saharan Africa |
|           Saint Helena           |     2020    |                99.00                |              100.00             |         None        | Sub-Saharan Africa |
+----------------------------------+-------------+-------------------------------------+---------------------------------+---------------------+--------------------+

### Task 3

In [ ]:
%%sql

SELECT 
	Country_name,
	Time_period,
	Pct_managed_drinking_water_services,
	Pct_managed_sanitation_services,
	Est_gdp_in_billions,
    	region
FROM 
	united_nations.Access_to_Basic_Services
WHERE
	region = 'Sub-Saharan Africa'
AND 
	Time_period = 2020
AND 
	Est_gdp_in_billions IS NOT NULL;

# Use LIMIT if you think the results set will be large
# Use ORDER BY Est_gdp_in_billions to order your results

### Task 4

Arrange the data in descending order. Comparing Nigeria to South Africa.  We see that Nigeria has a lower percentage in the availability of both water and sanitation services. 


In [ ]:
%%sql

SELECT 
	Country_name,
	Time_period,
	Pct_managed_drinking_water_services,
	Pct_managed_sanitation_services,
	Est_population_in_millions,
	Est_gdp_in_billions
FROM 
	united_nations.Access_to_Basic_Services

WHERE
	Region = ‘Sub-Saharan Africa
AND 
Time_period = 2020

AND 
	Est_gdp_in_billions IS NOT NULL

AND 
	Country_name  NOT IN ('Nigeria','South Africa','Ethiopia','Kenya','Ghana');


# Use LIMIT if you think the results set will be large
# Use ORDER BY Est_gdp_in_billions to order your results


### Task 5

Looking at the first entry without ordering any data it is interesting to note that Burundi, with a GDP of only 2.65 Billion has a similar percentage of managed services compared to Nigeria seen in our previous  query(77% and 42%). Nigeria had a percentage of 77 for managed water and 42 for managed sanitation. Arranging the percentage managed drinking water services in descending order.Mauritius with a GDP lower than that of Botswana has better drinking water and sanitation services. 

In [ ]:
%%sql

SELECT 
	Country_name,
	Time_period,
	Pct_managed_drinking_water_services,
	Pct_managed_sanitation_services,
	Est_population_in_millions,
	Est_gdp_in_billions
FROM 
	united_nations.Access_to_Basic_Services

WHERE
	Region = ‘Sub-Saharan Africa
AND 
Time_period = 2020

AND 
	Est_gdp_in_billions IS NOT NULL

AND 
	Country_name  IS IN ('Nigeria','South Africa','Ethiopia','Kenya','Ghana');

# Use LIMIT if you think the results set will be large
# Use ORDER BY Est_gdp_in_billions to order your results by GDP
# Use ORDER BY Pct_managed_drinking_water_services to order percentage managed drinking water services
# Use ORDER BY Pct_managed_sanitation_services to order your results by percentage managed sanitation services


## Summary

In this exercise we used the IS NULL statement to determine if there were any null values in the GDP column. 
We then used the IS NOT NULL statement to exclude those nulls.
We looked at the top 5 GDP's in Sub-Saharan Africa by using the IS IN statement. 
We had a look at the rest of the Sub-Saharan African countries by excluding the top 5 GDP's using the IS NOT IN statement
Based on the quick perusal of the data in Sub-Saharan africa, we can conclude that there isn’t
any noticeable correlation between GDP and the availability of drinking water and sanitation services. 


#  

<div align="center" style=" font-size: 80%; text-align: center; margin: 0 auto">
<img src="https://raw.githubusercontent.com/Explore-AI/Pictures/refs/heads/master/ALX_banners/ALX_Navy.png"  style="width:100px"  ;/>
</div>